modules/base/rendering/renderablemodel.cpp

```diff
@@ -279,103 +282,173 @@ RenderableModel::RenderableModel(const ghoul::Dictionary& dictionary)
+                    break;
+                case Parameters::ScaleUnit::Kilometer:
+                    _modelScale = DistanceUnit::Kilometer;
+                    break;
+                default:
+                    throw ghoul::MissingCaseException();
+            }
+            _scaleVector = glm::dvec3(convertUnit(_modelScale, DistanceUnit::Meter));
         }
-        else if (dictionary.hasValue<ghoul::Dictionary>(KeyGeomModelFile)) {
-            LWARNING("Loading a model with several files is deprecated and will be "
-                "removed in a future release"
-            );
+        else if (std::holds_alternative<float>(*p.modelScale)) {
+            _scaleVector = glm::dvec3(std::get<float>(*p.modelScale));
+        }
+    }
 
-            ghoul::Dictionary fileDictionary = dictionary.value<ghoul::Dictionary>(
-                KeyGeomModelFile
-            );
-            std::vector<std::unique_ptr<ghoul::modelgeometry::ModelGeometry>> geometries;
-
-            for (std::string_view k : fileDictionary.keys()) {
-                // Handle each file
-                file = absPath(fileDictionary.value<std::string>(k));
-                geometries.push_back(ghoul::io::ModelReader::ref().loadModel(
-                    file,
-                    ghoul::io::ModelReader::ForceRenderInvisible(_forceRenderInvisible),
-                    ghoul::io::ModelReader::NotifyInvisibleDropped(_notifyInvisibleDropped)
-                ));
-            }
+    if (p.animationStartTime.has_value()) {
+        if (!_geometry->hasAnimation()) {
+            LWARNING("Animation start time given to model without animation");
+        }
+        _animationStart = *p.animationStartTime;
+    }
 
-            if (!geometries.empty()) {
-                std::unique_ptr<ghoul::modelgeometry::ModelGeometry> combinedGeometry =
-                    std::move(geometries[0]);
-
-                 // Combine all models into one ModelGeometry
-                 for (unsigned int i = 1; i < geometries.size(); ++i) {
-                    for (ghoul::io::ModelMesh& mesh : geometries[i]->meshes()) {
-                        combinedGeometry->meshes().push_back(
-                            std::move(mesh)
-                        );
-                    }
-
-                    for (ghoul::modelgeometry::ModelGeometry::TextureEntry& texture :
-                        geometries[i]->textureStorage())
-                    {
-                        combinedGeometry->textureStorage().push_back(
-                            std::move(texture)
-                        );
-                    }
-                }
-                 _geometry = std::move(combinedGeometry);
-                _geometry->calculateBoundingRadius();
+    if (p.enableAnimation.has_value()) {
+        if (!_geometry->hasAnimation()) {
+            LWARNING("Attempting to enable animation for a model that does not have any");
+        }
+        else if (*p.enableAnimation &&_animationStart == "") {
```

Comment:
Nitpicky:  missing space after the `&&`

For std::string it's better to check `_animationStart.empty()`  as that way the `""` doesn't need to be converted to a `std::string` that then gets passed to the `operator==` function